MNE/disp3D/engine/model/items/common/abstract3Dtreeitem.cpp

```diff
@@ -0,0 +1,314 @@
+* PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+* NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+*
+* @brief    Abstract3DTreeItem class definition.
+*
+*/
+//*************************************************************************************************************
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include "abstract3Dtreeitem.h"
+#include "../common/metatreeitem.h"
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// Qt INCLUDES
+//=============================================================================================================
+
+#include <Qt3DExtras/QPhongMaterial>
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// Eigen INCLUDES
+//=============================================================================================================
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace DISP3DLIB;
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+Abstract3DTreeItem::Abstract3DTreeItem(QEntity* p3DEntityParent, int iType, const QString& text)
+: QStandardItem(text)
+, Renderable3DEntity(p3DEntityParent)
+, m_iType(iType)
+{
+    initItem();
+}
+
+
+//*************************************************************************************************************
+
+void Abstract3DTreeItem::initItem()
+{
+    this->setToolTip("Abstract 3D Tree Item");
+
+    //Transformation
+    QList<QStandardItem*> list;
+    QVariant data;
+
+    if(!m_pItemTransformationOptions) {
+        m_pItemTransformationOptions = new MetaTreeItem(MetaTreeItemTypes::UnknownItem, "Transformation");
```

Comment:
MetraTreeItems are inheriting QStandardItem. Qt internally handles deletion of its items. No need for shared pointers here. Also, QStandardItem do not accept new row or coulm items passed as shared pointers. Qpointer is the best we can do here.